### DISCUSSION TITLE
ok.ru

### DISCUSSION THREAD

**Discussion by HalowTV** [streamlink/streamlink#5377] [General]
### Description

Hi i runt ubuntu 18.04 server  

hi, I run like this the video works but my problem is the video works for only 24 hours then I have to update if all the time is there any other way that it can update without me having to do it?
nohup streamlink -p mpv "https://ok.ru/video/3774185147064" best --player-external-http --player-external-http-port 184 &

**Comment by andaluspc** [comment]
Hello,
your video is now deleted I believe. but I went to ok.ru and pickedup a random video. of the following URL:  https://ok.ru/video/6815852857886

when you do command ( streamlink https://ok.ru/video/6815852857886 --stream-url), you'll get the stream URL that broadcasts actual stream. in my case it was: 
https://vd340.mycdn.me/video.m3u8?cmd=videoPlayerCdn&expires=1693594025962&srcIp=188.123.191.63&srcAg=GECKO&ms=185.226.52.61&type=4&sig=qPGKy8xVrBQ&ct=8&urls=45.136.20.16&clientType=0&id=5090762033694

Noteis the 'expires' word within the URL followed by a number which is actually the date/time when the video will expire measured n epoch time. I found it is exactly 24 hours from the time you runt he streamlink command.

What I usually do is to schedule to run a command via a script every, say, 12 hours and get the new URL embedded into my m3u8 file.

**Comment by leomani51214** [comment]
Hello,
I am using the following Streamlink command to extract the live stream URL for a broadcast that I created on ok.ru :
streamlink "https://ok.ru/video/8532062379616" --stream-url
This successfully returns a .m3u8 link for the stream : 
https://vsd242.okcdn.ru/hls/7390660069984.m3u8/sig/iPJ8iZErCo/expires/1735602941172/srcIp/39.63.235.143/urls/45.136.20.58/clientType/0/srcAg/GECKO/mid/8532062379616/video.m3u8?p
However, there is an issue with the validity of the URL. Even though I can extract this .m3u8 link, it doesn't play after one minute of extraction. Upon decoding the expires parameter in the URL, I discovered that the link has an expiration date set to July 2024, while the current date is December 2024.

This is the primary reason why the stream doesn't work as expected.

Question:

Can anyone help me understand why the link isn't working, and provide guidance on how to refresh or extend the expiration time so that it works correctly? Specifically, I need a way to ensure that the stream link remains valid for a longer period of time.
